src/qt3utils/applications/controllers/andor_spectrometer.py

```diff
@@ -0,0 +1,988 @@
+    # - Turret
+    grating: tk.StringVar
+    center_wavelength: tk.DoubleVar
+
+    # - Calibration
+    pixel_offset: tk.DoubleVar
+    wavelength_offset: tk.DoubleVar
+    # - Ports
+    input_port: tk.StringVar
+    output_port: tk.StringVar
+
+    # Acquisition
+    # - Modes
+    read_mode: tk.StringVar
+    acquisition_mode: tk.StringVar
+    trigger_mode: tk.StringVar
+    # - Timing
+    exposure_time: tk.DoubleVar
+    number_of_accumulations: tk.IntVar
+    accumulation_cycle_time: tk.DoubleVar
+    number_of_kinetics: tk.IntVar
+    kinetic_cycle_time: tk.DoubleVar
+    #  - Data-Pre-Processing
+    baseline_clamp: tk.BooleanVar
+    cosmic_ray_removal: tk.BooleanVar
+    keep_clean_on_external_trigger: tk.BooleanVar
+    # - Single Track Setup
+    single_track_center_row: tk.IntVar
+    single_track_height: tk.IntVar
+
+    # Electronics
+    # - Vertical Shift
+    vertical_shift_speed: tk.StringVar
+    # - Horizontal Shift
+    horizontal_shift_speed: tk.StringVar
+    pre_amp_gain: tk.StringVar
+
+    # Temperature
+    # - Set Point
+    target_sensor_temperature: tk.IntVar
+    reach_temperature_before_acquisition: tk.BooleanVar
+    # - Cooler
+    cooler: tk.BooleanVar
+    cooler_persistence: tk.BooleanVar
+
+    def update_variables_from_dict(self, config_dict: Dict[str, Any]) -> None:
+        """
+        Updates the variables from the given configuration dictionary.
+
+        Parameters
+        ----------
+        config_dict: Dict[str, float]
+            The configuration dictionary.
+        """
+        for key in config_dict.keys():
+            if hasattr(self, key):
+                variable: tk.Variable = getattr(self, key)
+                variable.set(config_dict[key])
+            else:
+                message = (f"Unknown key '{key}' was passed in Andor "
+                           f"Spectrometer configuration dictionary.")
+                self.logger.warning(message)
+
+    def get_config_dict(self) -> Dict[str, Any]:
+        """
+        Returns the configuration dictionary.
+
+        Returns
+        -------
+        Dict[str, Any]
+            The configuration dictionary.
+        """
+        variable_keys = [f.name for f in fields(self) if isinstance(getattr(self, f.name), tk.Variable)]
+
+        config_dict = {}
+        for key in variable_keys:
+            variable: _TkVarType = getattr(self, key)
+            config_dict[key] = variable.get()
+
+        return config_dict
+
+    def variable_dict(self) -> Dict[str, _TkVarType]:
+        """
+        Returns a dictionary of the GUI variables
+
+        Returns
+        -------
+        Dict[str, _TkVarType]
+            A dictionary of the GUI variables.
+        """
+        variable_keys = [f.name for f in fields(self) if isinstance(getattr(self, f.name), tk.Variable)]
+
+        var_dict = {}
+        for key in variable_keys:
+            var_dict[key] = getattr(self, key)
+
+        return var_dict
+
+
+class ConfigurationView:
```

Comment:
Thanks for the doctring. I have a rough idea of what this does and I am curious, does it make sense to have this class in this file? If this is a window popup I feel like we would want to put it elsewhere, just not sure where though

Reply:
Since this class is related only to the andor spectrometer controller, I think it makes sense to be in this file